fix unit test failure in ./pkg/api/testing/serialization_test.go

WanLinghao opened the pull request:
> /kind failing-test


**What this PR does / why we need it**:
fix unit test failure in ./pkg/api/testing/serialization_test.go with error like:
```
--- FAIL: TestSetControllerConversion (0.00s)
    serialization_test.go:104: rs._internal.apps -> rs.v1.apps
    serialization_test.go:107: unexpected encoding error: no kind "ReplicaSet" is registered for version "apps/v1beta1" in scheme "k8s.io/kubernetes/pkg/api/legacyscheme/scheme.go:29"
```
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #71369 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
<!--  
If no, just write "NONE".
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".
2. 
-->
```release-note
NONE
```

WanLinghao commented:
/retest

neolit123 commented:
/kind failing-test
/priority important-longterm

neolit123 commented:
/retest

liggitt commented:
this issue is due to the unit tests that use `testapi` not being deterministic (they are affected by envvars set in the test environment that change API group order).

rather than shuffling order around, can we remove use of testapi from this test and pin the tested version to apps/v1?

jennybuckley commented:
/cc @jpbetz

WanLinghao committed:
fix unit test failure in ./pkg/api/testing/serialization_test.go with error like:

    --- FAIL: TestSetControllerConversion (0.00s)
        serialization_test.go:104: rs._internal.apps -> rs.v1.apps
        serialization_test.go:107: unexpected encoding error: no kind "ReplicaSet" is registered for version "apps/v1beta1" in scheme "k8s.io/kubernetes/pkg/api/legacyscheme/scheme.go:29"

WanLinghao commented:
/test pull-kubernetes-integration

WanLinghao commented:
@liggitt comments addressed, PTAL

WanLinghao commented:
@liggitt comment addressed, PTAL

liggitt commented:
/lgtm

liggitt commented:
/approve
